pkg/gui/controllers/files_controller.go

```diff
@@ -684,6 +690,26 @@ func (self *FilesController) Open() error {
 	return self.c.Helpers().Files.OpenFile(node.GetPath())
 }
 
+func (self *FilesController) openDiffTool(node *filetree.FileNode) error {
+	fromCommit := ""
+	reverse := false
+	if self.c.Modes().Diffing.Active() {
+		fromCommit = self.c.Modes().Diffing.Ref
+		reverse = self.c.Modes().Diffing.Reverse
+	}
+	return self.c.RunSubprocessAndRefresh(
+		self.c.Git().Diff.OpenDiffToolCmdObj(
+			git_commands.DiffToolCmdOptions{
+				Filepath:    node.Path,
+				FromCommit:  fromCommit,
+				ToCommit:    "",
+				Reverse:     reverse,
+				IsDirectory: !node.IsFile(),
+				Staged:      !node.GetHasUnstagedChanges(),
```

Comment:
The logic here is different from what we're doing when copying a diff: for that, we use the staged changes if there are any, and the unstaged changes otherwise. Here, we use the staged changes only if there are no unstaged changes.

This code is so old that I don't totally remember why I did it like this, but I'm sure I had a reason; I found it more likely that you want to look at the unstaged changes than the staged ones. Maybe it was just because that's the default of `git difftool` on the command line when you call it with no options.